src/Mod/PartDesign/Gui/TaskHoleParameters.cpp

```diff
@@ -798,9 +796,17 @@ void TaskHoleParameters::changedObject(const App::Document&, const App::Property
         widget->setDisabled(ro);
     };
 
-    if (&Prop == &hole->Threaded) {
-        ui->Threaded->setEnabled(true);
-        updateCheckable(ui->Threaded, hole->Threaded.getValue());
+    if (&Prop == &hole->Threaded || &Prop == &hole->ModelThread) {
+        if (hole->Threaded.getValue()) {
+            if (hole->ModelThread.getValue()) {
+                updateComboBox(ui->HoleType, static_cast<int>(HoleTypeIndex::ModeledThread));
+            } else {
+                updateComboBox(ui->HoleType, static_cast<int>(HoleTypeIndex::TapDrill));
+            }
+        } else {
+            updateComboBox(ui->HoleType, static_cast<int>(HoleTypeIndex::Clearance));
+        }

```

Comment:
This code is repeated - if you extract it to a private function like I suggested in comment above it should simplify that case too.